File: programmers/level1/k번째수.py
def solution(array, commands):
    ans = []
    for c in commands:
        a = array[c[0]-1:c[1]].sort()
        a.sort()
        ans.append(a[c[2]-1])
    
    return ans


if __name__ == "__main__":
    array = [1, 5, 2, 6, 3, 7, 4]
    commands = [[2, 5, 3], [4, 4, 1], [1, 7, 3]]
    print(solution(array, commands))

File: python/programmers/level1/k번째수.kt
package test

class KthNumber {
    fun solution(array: IntArray, commands: Array<IntArray>): IntArray {
        val answer = mutableListOf<Int>()

        for (c in commands) {
            val a = array.slice(IntRange(c[0]-1, c[1]-1)).sorted()
            answer.add(a[c[2]-1])
        }

        return answer.toIntArray()
    }
}

fun main() {
    val arr1 = intArrayOf(1, 5, 2, 6, 3, 7, 4)
    val arr2 = arrayOf(
        intArrayOf(2, 5, 3),
        intArrayOf(4, 4, 1),
        intArrayOf(1, 7, 3)
    )
    println(KthNumber().solution(arr1, arr2))
}